contracts/examples/mintable-erc1155-transfer/FxMintableERC1155ChildTunnel.sol

```diff
@@ -0,0 +1,261 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {FxERC1155} from "../../tokens/FxERC1155.sol";
+import {FxBaseChildTunnel} from "../../tunnel/FxBaseChildTunnel.sol";
+import {Ownable} from "../../lib/Ownable.sol";
+import {Address} from "../../lib/Address.sol";
+import {ERC1155Holder} from "../../lib/ERC1155Holder.sol";
+import {Create2} from "../../lib/Create2.sol";
+
+contract FxMintableERC1155ChildTunnel is FxBaseChildTunnel, Create2, ERC1155Holder, Ownable {
+    bytes32 public constant DEPOSIT = keccak256("DEPOSIT");
+    bytes32 public constant DEPOSIT_BATCH = keccak256("DEPOSIT_BATCH");
+    bytes32 public constant WITHDRAW = keccak256("WITHDRAW");
+    bytes32 public constant WITHDRAW_BATCH = keccak256("WITHDRAW_BATCH");
+
+    event TokenMapped(address indexed rootToken, address indexed childToken);
+    event FxWithdrawMintableERC1155(
+        address indexed rootToken,
+        address indexed childToken,
+        address indexed userAddress,
+        uint256 id,
+        uint256 amount
+    );
+    event FxDepositMintableERC1155(
+        address indexed rootToken,
+        address indexed depositor,
+        address indexed userAddress,
+        uint256 id,
+        uint256 amount
+    );
+    event FxBatchWithdrawMintableERC1155(
+        address indexed rootToken,
+        address indexed childToken,
+        address indexed userAddress,
+        uint256[] ids,
+        uint256[] amounts
+    );
+    event FxBatchDepositMintableERC1155(
+        address indexed rootToken,
+        address indexed depositor,
+        address indexed userAddress,
+        uint256[] ids,
+        uint256[] amounts
+    );
+    // root to child token
+    mapping(address => address) public rootToChildToken;
+    // child token template
+    address public immutable childTokenTemplate;
+    // root token template codehash
+    bytes32 public immutable rootTokenTemplateCodeHash;
+
+    constructor(
+        address _fxChild,
+        address _childTokenTemplate,
+        address _rootTokenTemplate
+    ) FxBaseChildTunnel(_fxChild) {
+        childTokenTemplate = _childTokenTemplate;
+        require(
+            Address.isContract(_childTokenTemplate),
+            "FxMintableERC1155ChildTunnel: Token template is not contract"
+        );
+        // compute root token template code hash
+        rootTokenTemplateCodeHash = keccak256(minimalProxyCreationCode(_rootTokenTemplate));
+    }
+
+    //
+    // External methods
+    //
+
+    // deploy child token with unique id
+    function deployChildToken(uint256 _uniqueId, string calldata _uri) external {
```

Comment:
Shouldn't the access of this function be restricted?

Reply:
I get the point that why it shouldn't be restricted, but then it should be removed from other contracts as well (`FxMintableERC20ChildTunnel` & `FxMintableERC721ChildTunnel`) otherwise it will be a specification mismatch.